### PR TITLE
Make switch widget accept 3 other colors.

### PR DESCRIPTION
**Pull request by jiajiabingcheng**
Add 3 arguments: inactiveThumbColor, inactiveTrackColor, activeTrackColor to Switch widget. So developers can pass in custom color.

### CODE REVIEW

File: packages/flutter/lib/src/material/switch.dart [hunk 1]
```diff
@@ -46,14 +46,17 @@ class Switch extends StatefulWidget {
...
+      {Key key,
+      @required this.value,
+      @required this.onChanged,
+      this.activeColor,
+      this.activeTrackColor,
+      this.inactiveThumbColor,
+      this.inactiveTrackColor,
+      this.activeThumbImage,
+      this.inactiveThumbImage})
+      : super(key: key);
```
Comment (Hixie) [inline packages/flutter/lib/src/material/switch.dart]: nit: please use the original style, see https://github.com/flutter/flutter/wiki/Style-guide-for-Flutter-repo for details on our style

File: packages/flutter/lib/src/material/switch.dart [hunk 2]
```diff
@@ -89,6 +92,37 @@ class Switch extends StatefulWidget {
   /// Defaults to accent color of the current [Theme].
   final Color activeColor;
 
+  /// The color to use on the track when this switch is on.
+  ///
+  /// Defaults to accent color of the current [Theme] withAlpha(0x80).
```
Comment (Hixie) [inline packages/flutter/lib/src/material/switch.dart]: "Defaults to accent color of the current [Theme], with the opacity set at 50%."

File: packages/flutter/lib/src/material/switch.dart [hunk 3]
```diff
@@ -89,6 +92,37 @@ class Switch extends StatefulWidget {
...
+  /// Defaults to:
+  /// if the brightness of current [Theme] is dark and [onChanged] callback is
+  /// not null: white30;
+  /// if the brightness of current [Theme] is dark and [onChanged] callback is
+  /// null: white10;
+  /// if the brightness of current [Theme] is not dark and [onChanged] callback
+  /// is not null: black26;
+  /// if the brightness of current [Theme] is not dark and [onChanged] callback
+  /// is null: black12.
+  final Color inactiveTrackColor;
```
Comment (Hixie) [inline packages/flutter/lib/src/material/switch.dart]: these two comments will each end up all jumbled into one paragraph.

We should probably just say "Defaults to the colors described in the Material design specification" or some such. I doubt readers are going to care about exactly what shade of white it happens to default to.

File: packages/flutter/lib/src/material/switch.dart [hunk 4]
```diff
@@ -114,16 +148,21 @@ class _SwitchState extends State<Switch> with TickerProviderStateMixin {
     final bool isDark = themeData.brightness == Brightness.dark;
 
     final Color activeThumbColor = widget.activeColor ?? themeData.accentColor;
-    final Color activeTrackColor = activeThumbColor.withAlpha(0x80);
+    final Color activeTrackColor =
+        widget.activeTrackColor ?? activeThumbColor.withAlpha(0x80);
```
Comment (Hixie) [inline packages/flutter/lib/src/material/switch.dart]: we should wrap this in a manner consistent with the cases below

**Comment by Hixie** [comment]
Looks good! Can you add a test as well?

**Comment by jiajiabingcheng** [comment]
Hi Hixie,

Thanks for the review!

For the test, the state for this 'Switch' widget is a private class, so it's hard to test what color it actually takes after the widget is built up. I took an example of the title bar test (https://github.com/flutter/flutter/blob/e57d94c3088a66cb58a78ac4ad4d2997a1623147/packages/flutter/test/widgets/app_title_test.dart) and tested the switch can take the four custom colors.

**Comment by Hixie** [comment]
Best way to test this is probably to use the `mock_canvas.dart` API. Search for other tests, and look at the way they do `expect(foo, paints...)`

**Comment by jiajiabingcheng** [comment]
Thanks Hixie! Sorry for the delay. I just added the test.

**Comment by Hixie** [comment]
Looks great, thanks for your contribution!

**Pull request merged**